### PR TITLE
PIM-9889: Merge 5.0 into master

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php [hunk 1]
```diff
@@ -58,6 +59,10 @@ class PublishJobToQueue
     /** @var EventDispatcherInterface */
     private $eventDispatcher;
 
+    /** @var BatchLogHandler */
+    private $batchLogHandler;
+
+    // TODO @merge master/6.0: make the $batchLogHandler argument mandatory
```
Comment (rouliane) [inline src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php:65]: Todo to handle

File: src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php [hunk 2]
```diff
@@ -119,6 +126,11 @@ public function publish(
         $this->validateJobParameters($jobInstance, $jobParameters, $jobInstanceCode);
         $jobExecution = $this->jobRepository->createJobExecution($jobInstance, $jobParameters);
 
+        // TODO @merge master/6.0: remove the null !== $this->>batchLogHandler check
```
Comment (rouliane) [inline src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php:129]: todo to handle

**Review by rouliane — APPROVED** [review]

**Review by franckbrouard — APPROVED** [review]

File: src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php (hunk 1 above)
Comment (anaisbaunelemaire) [reply src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php:65]: My bad, I misunderstood this one..!

**Review by mmetayer — APPROVED** [review]

**Comment by anaisbaunelemaire** [comment]
ok

**Pull request merged**